src/main/java/flight/reservation/order/FlightOrder.java

```diff
@@ -40,48 +40,34 @@ private boolean isOrderValid(Customer customer, List<String> passengerNames, Lis
         return valid;
     }
 
-    public boolean processOrderWithCreditCardDetail(String number, Date expirationDate, String cvv) throws IllegalStateException {
+    public void processOrderWithCreditCardDetail(String number, Date expirationDate, String cvv) throws IllegalStateException {
         CreditCard creditCard = new CreditCard(number, expirationDate, cvv);
-        return processOrderWithCreditCard(creditCard);
+        processOrderWithCreditCard(creditCard);
     }
 
-    public boolean processOrderWithCreditCard(CreditCard creditCard) throws IllegalStateException {
+    public void processOrderWithCreditCard(CreditCard creditCard) throws IllegalStateException {
         if (isClosed()) {
             // Payment is already proceeded
-            return true;
+            return;
         }
-        // validate payment information
-        if (!cardIsPresentAndValid(creditCard)) {
-            throw new IllegalStateException("Payment information is not set or not valid.");
-        }
-        boolean isPaid = payWithCreditCard(creditCard, this.getPrice());
-        if (isPaid) {
-            this.setClosed();
-        }
-        return isPaid;
+        payWithCreditCard(creditCard, this.getPrice());
```

Comment:
Missing exception handling for unpresent and/or invalid card. (see payWIthCreditCard(...) method).
This should actually set isPaid to false and terminate without an exception being thrown.